src/main/java/org/apache/camel/idea/CamelContributor.java

```diff
@@ -21,110 +21,50 @@
-                return; //we are done
-            }
-
-            String componentName = StringUtils.asComponentName(val);
-            if (componentName != null && camelCatalog.findComponentNames().contains(componentName)) {
-
-                // it is a known Camel component
-                String json = camelCatalog.componentJSonSchema(componentName);
-                ComponentModel componentModel = ModelHelper.generateComponentModel(json, true);
-
-                // grab all existing parameters
-                String query = val;
-                // strip up ending incomplete parameter
-                if (query.endsWith("&") || query.endsWith("?")) {
-                    query = query.substring(0, query.length() - 1);
-                }
-
-                Map<String, String> existing = null;
-                try {
-                    existing = camelCatalog.endpointProperties(query);
-                } catch (Exception e) {
-                    // ignore
-                }
-
-                // are we editing an existing parameter value
-                // or are we having a list of suggested parameters to choose among
-                boolean editSingle = val.endsWith("=");
-                boolean editQueryParameters = val.contains("?");
-                boolean editContextPath = !editQueryParameters;
-
-                List<LookupElement> answer = null;
-                if (editSingle) {
-                    // parameter name is before = and & or ?
-                    int pos = Math.max(val.lastIndexOf('&'), val.lastIndexOf('?'));
-                    String name = val.substring(pos + 1);
-                    name = name.substring(0, name.length() - 1); // remove =
-                    EndpointOptionModel endpointOption = componentModel.getEndpointOption(name);
-                    if (endpointOption != null) {
-                        answer = addSmartCompletionForSingleValue(parameters.getEditor(), val, suffix, endpointOption);
-                    }
-                } else if (editQueryParameters) {
-                    // suggest a list of options for query parameters
-                    answer = addSmartCompletionSuggestionsQueryParameters(val, componentModel, existing);
-                } else if (editContextPath) {
-                    // suggest a list of options for context-path
-                    answer = addSmartCompletionSuggestionsContextPath(val, componentModel, existing);
-                }
-
-                // are there any results then add them
-                if (answer != null && !answer.isEmpty()) {
-                    resultSet.withPrefixMatcher(val).addAllElements(answer);
-                }
-            }
+            camelCompletionExtensions.stream()
+                    .filter(p -> p.isValid(parameters,context,tuple))
+                    .forEach(p -> p.addCompletions(parameters,context,resultSet,tuple));
```

Comment:
I think we should not call multiple, eg it should be first found that is invoked only.